packages/create-app/src/lib/tasks.ts

```diff
@@ -236,3 +236,59 @@ export async function moveAppTask(
       });
   });
 }
+
+/**
+ * Checks if git package is installed and git credentials exists
+ *
+ * @throws if `exec` fails
+ */
+export async function checkForGitSetup(): Promise<boolean> {
+  const runCmd = (cmd: string) =>
+    exec(cmd).catch(error => {
+      process.stdout.write(error.stderr);
+      process.stdout.write(error.stdout);
+      throw new Error(`Could not execute command ${chalk.cyan(cmd)}`);
+    });
+
+  try {
+    await runCmd('which git');
+
+    const [gitUsername, gitEmail] = await Promise.all([
+      runCmd('git config user.name'),
+      runCmd('git config user.email'),
```

Comment:
While we're at it, perhaps we should grab `init.defaultBranch` here as well? Rather than printing it during initialization. That way I think we can get rid of the double `git init` as well, which is a lil awkward

Reply:
That's a good idea, gonna try it later on